sdk/keyvault/keyvault-admin/src/backupClient.ts

```diff
@@ -0,0 +1,277 @@
+   * const sasToken = "<sas-token>";
+   * const poller = await client.beginBackup(blobStorageUri, sasToken);
+   *
+   * // Serializing the poller
+   * const serialized = poller.toString();
+   * // A new poller can be created with:
+   * // await client.beginBackup(blobStorageUri, sasToken, { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const backupUri = await poller.pollUntilDone();
+   * console.log(backupUri);
+   * ```
+   * @summary Creates a new role assignment.
+   * @param blobStorageUri The URL of the blob storage resource, with the path to the folder name where the backup will end up being generated.
+   * @param sasToken The SAS token.
+   * @param [options] The optional parameters.
+   */
+  public async beginBackup(
+    blobStorageUri: string,
+    sasToken: string,
+    options: BeginBackupOptions = {}
+  ): Promise<PollerLike<PollOperationState<string>, string>> {
+    if (!(blobStorageUri && sasToken)) {
+      throw new Error(
+        "beginBackup requires non-empty strings for the parameters: blobStorageUri and sasToken."
+      );
+    }
+
+    const poller = new BackupPoller({
+      blobStorageUri,
+      sasToken,
+      client: this.client,
+      vaultUrl: this.vaultUrl,
+      intervalInMs: options.intervalInMs,
+      resumeFrom: options.resumeFrom,
+      requestOptions: options
+    });
+
+    // This will initialize the poller's operation (the generation of the backup).
+    await poller.poll();
+
+    return poller;
+  }
+
+  /**
+   * Starts restoring all key materials using the SAS token pointing to a previously stored Azure Blob storage
+   * backup folder.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indefinitely until the Key Vault backup is generated.
+   *
+   * Example usage:
+   * ```ts
+   * const client = new KeyVaultBackupClient(url, credentials);
+   *
+   * const blobStorageUri = "<blob-storage-uri>"; // <Blob storage URL>/<folder name>
+   * const sasToken = "<sas-token>";
+   * const folderName = "<folder-name>";
+   * const poller = await client.beginRestore(blobStorageUri, sasToken, folderName);
+   *
+   * // Serializing the poller
+   * const serialized = poller.toString();
+   * // A new poller can be created with:
+   * // await client.beginRestore(blobStorageUri, sasToken, folderName, { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const backupUri = await poller.pollUntilDone();
+   * console.log(backupUri);
+   * ```
+   * @summary Creates a new role assignment.
+   * @param blobStorageUri The URL of the blob storage resource, with the folder name of the blob where the previous successful full backup was stored.
+   * @param sasToken The SAS token.
+   * @param folderName The folder name of the blob where the previous successful full backup was stored.
+   * @param [options] The optional parameters.
+   */
+  public async beginRestore(
```

Comment:
python has `begin_full_restore`, and tbh I think this naming is better, since we have selective restore

Reply:
I can do that!

Reply:
Wait, .Net has startRestore and startSelective restore: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Administration/api/Azure.Security.KeyVault.Administration.netstandard2.0.cs#L52-L55

So, before doing this change, I want us to coordinate 🤔 @christothes , @heaths

Reply:
I'm open to snapping to `*FullRestore`, though I don't feel strongly either way.

Reply:
In general, I'd say the naming inconsistencies can all be rationalized for beta2.

Reply:
agreed, apologies should've been more clear in this PR review that this isn't blocking